Indentifiaction for user

vphilippon opened the pull request:
@grondinjc @maym2104 Ready for review.

vphilippon committed:
Basic User login base

vphilippon committed:
Use App/Controller structure for Identifier

vphilippon committed:
Rename Identifier -> Identify

vphilippon committed:
Change 'conditions' for a single implicit requirement

vphilippon committed:
Look at sessions for used username

grondinjc commented on src/server/identifyController.py:
```diff
@@ -0,0 +1,126 @@
+  is not None, a login is required if user is not identified already.
+  """
+  login_required = cherrypy.request.config.get('identify.require', False)
+  from_page = urllib.quote(cherrypy.request.request_line.split()[1])
```
@vphilippon  This line could be moved to the else section before the raise redirect since it is only needed there

vphilippon replied:
@grondinjc Yep, that's right

grondinjc commented on src/server/identifyController.py:
```diff
@@ -0,0 +1,126 @@
+      %(msg)s<br />
+      Username: <input type="text" name="username" value="%(username)s" /><br />
+      <input type="submit" value="Log in" />
+    </body></html>""" % locals()
```
@vphilippon  Could this html be moved into a template ?

vphilippon replied:
@grondinjc Yes, this was temporary.

grondinjc commented on src/server/identifyController.py:
```diff
@@ -0,0 +1,126 @@
+      cherrypy.request.login = None
+      self._app.removeUsername(username)
+
+    raise cherrypy.HTTPRedirect(from_page or "/")
```
@vphilippon  The defaut from page is / 
Could it be moved into a constant at the top of the file since it is repeated 4 times ?

vphilippon replied:
Done

grondinjc commented on bin/configs/server.conf.example:
```diff
@@ -4,7 +4,9 @@ server.socket_host='0.0.0.0'
 server.socket_port=8080
 server.thread_pool=20
 tools.sessions.on=True
+tools.sessions.clean_freq=2
```
@vphilippon  What is it for ?
Could you had a comment to explain its purpose ?

vphilippon replied:
@grondinjc I will add a comment.
The point is that once a session expire, it's not removed from the sessions cache right away. So, when we're looking at the sessions to check if the username is already in use, the expired session are still there. The clean_freq=2 means "every 2 minutes, clear the expired session"
So, if your session expire it'll take a maximum of 2 minutes before you can log with the same user again.

vphilippon committed:
Constant for redirect + comment and misc corrections

vphilippon committed:
Use Template for login

vphilippon committed:
PEP8 + Redirect to '/ide' + @require_identify for all in IDEController

vphilippon committed:
WS attemps to reconnect correction.

We may want to change for location.reload() instead, since
identification may have been lost...

vphilippon commented:
@grondinjc Changes done
